rules/CodeQuality/Rector/ClassConstFetch/ReplaceProductionConstantsWithLiteralsInTestFilesRector.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\ClassConstFetch;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use Rector\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Rector\AbstractRector;
+use Rector\Skipper\Matcher\FileInfoMatcher;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+use function assert;
+
+/**
+ * @see \Rector\Tests\CodeQuality\Rector\ClassConstFetch\ReplaceProductionConstantsWithLiteralsInTestFilesRector\ReplaceProductionConstantsWithLiteralsInTestFilesRectorTest
+ */
+class ReplaceProductionConstantsWithLiteralsInTestFilesRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    private const ALLOWED_PATTERNS = 'allowedPatterns';
+
+    private const ALLOWED_CONSTANTS = 'allowedConstants';
+
+    /**
+     * @var array<string>
+     */
+    private array $allowedPatterns = [];
+
+    /**
+     * @var array<string>
+     */
+    private array $allowedConstants = [];
+
+    public function __construct(
+        private readonly ReflectionProvider $reflectionProvider,
+        private readonly FileInfoMatcher $fileInfoMatcher,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace production constants with literals in test files',
+            [
+                new CodeSample(
```

Comment:
use `ConfiguredCodeSample` class for configurable rules, see example

https://github.com/rectorphp/rector-src/blob/a2b010d3152c065aeb91bdbcd5437b8645a3f7c7/rules/Arguments/Rector/ClassMethod/ArgumentAdderRector.php#L60